erpnext/hr/doctype/employee_checkin/employee_checkin.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import datetime, time
+import frappe
+from frappe.model.document import Document
+
+class EmployeeCheckin(Document):
+	def validate(self):
+		if self.out_time:
+			in_time = datetime.datetime(*time.strptime(str(self.in_time), "%Y-%m-%d %H:%M:%S")[:6])
+			out_time = datetime.datetime(*time.strptime(str(self.out_time), "%Y-%m-%d %H:%M:%S")[:6])
+			if in_time > out_time:
+				frappe.throw("Out-time must be later than In-time.")
+			else:
+				duration = out_time - in_time
+				self.duration = duration
+
+@frappe.whitelist()
+def punch_in(rfid_tag):
```

Comment:
There is a security issue here. We need to restrict this API only to a particular role (Auto Attendance), else someone might figure out a way to punch in via a simple http request